Doc/library/sqlite3.rst

```diff
@@ -40,51 +40,48 @@ This document includes four main sections:
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+To use the module, start by opening a connection to an SQLite database.
+We do this by using the :func:`sqlite3.connect` function.
+The returned :class:`Connection` object represents our
+connection to the database.
+In our example,
+the database will be stored in the file :file:`example.db`::
 
    import sqlite3
    con = sqlite3.connect('example.db')
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
-
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+Notice that the file :file:`example.db` will be created implicitly
+if it does not exist.
+Now, create a :class:`Cursor` object using :meth:`~Connection.cursor`.
```

Comment:
I'm not keen on drifting too far away from the core of the tutorial: learning the basic API of the sqlite3 module. Perhaps we can link to an external resource? If Sphinx lets us create tooltips, that might be an option.

Reply:
As for the transaction below, I don't expect an indepth explanation.

I think the issue is mostly that the tutorial is telling me to do something (that I might not fully understand) with no justification of *why* we are doing it and what is it going to accomplish.

Even a simple sentence like "In order to execute SQL queries, we need to create a Cursor object using conn.cursor." would go a long way IMHO.

Reply:
Thanks, both; I'm convinced :) I'll revise it later today.

As for the SQL itself: I'll try to come up with a more concise example than the stocks thing. Preferrably some table with two, max three, rows. Maybe the old personnel database example, or even the music catalogue example should be used, as they will be familiar to folks who've done an SQL tutorial or two.

Reply:
I would pick a topic that most people like and are familiar with.  Music would be a good choice IMHO, or perhaps animals, planets, programming languages.  I would create at least 3 rows/columns.

Reply:
I went for Python release management :) Release dates, etc. I can expand from two to three columns.

Reply:
I never knew Python releases could be so fun; the release managers must be really lucky to have that job :laughing:

Reply:
Yeah... very fun 😆 Ok, ok, let's do planets instead. We'll let the first inserted row be Pluto; it'll be fun to watch the bug tickets: "Bug in sqlite3 tutorial: Pluto is not a planet"

Reply:
More seriously: maybe Python releases is not the best example :) I'm fine with anything. Perhaps the music catalogue is the way to go.

Reply:
Well, it isn't...unless you're also inserting Eris, Vesta, etc.

Trust me, I'm a NASA <sub><sub>*-funded earth*</sub></sub> scientist

Reply:
Ok, both fun _and_ pythonic: a Monty Python film (or skit, or quotes) database.

Reply:
Monty Python movies SGTM.  As columns you can have title, release date, and review score (you can pick imdb or rt scores, but you need to mention them).  Both the subject (movies) and the fields (title, date, score) are simple and familiar to people, it's pythonic, and as an added bonus it might make people curious about them.

Reply:
SGTM too